src/main/java/de/tum/in/www1/artemis/service/icl/SshGitLocationResolverService.java

```diff
@@ -71,13 +71,19 @@ public Path resolveRootDirectory(String command, String[] args, ServerSession se
 
         // git-upload-pack means fetch (read operation), git-receive-pack means push (write operation)
         final var repositoryAction = gitCommand.equals("git-upload-pack") ? RepositoryActionType.READ : gitCommand.equals("git-receive-pack") ? RepositoryActionType.WRITE : null;
-        final var user = session.getAttribute(SshConstants.USER_KEY);
-        try {
-            localVCServletService.authorizeUser(repositoryTypeOrUserName, user, exercise, repositoryAction, localVCRepositoryUri.isPracticeRepository());
+
+        if (session.getAttribute(SshConstants.IS_BUILD_AGENT_KEY) && repositoryAction == RepositoryActionType.READ) {
+            // We already checked for build agent authenticity
         }
```

Comment:
Wouldn't limiting the scope a build agent can access also make sense? As we can now uniquely identify the build agent, we could also check that the repo belongs to one of the agent's active jobs.

Might be worth considering for a follow-up PR.

Reply:
I think it’s very hard to do this properly. Just checking the exercise itself is not sufficient, as the solutions are also confidential.
On the other hand, there might be additional auxiliary repositories that have to be checked as well.
If you want very fine-grained permissions, we would probably have to come up with some access tokens only valid for one clone operation of a specific repository.
So definitely out of scope for now.